### PR TITLE
chore: move to plenary.path

### PR DESCRIPTION
**Pull request by Conni2461**
Close https://github.com/nvim-telescope/telescope-frecency.nvim/issues/23

CC @clason

### CODE REVIEW

File: lua/telescope/_extensions/frecency.lua [hunk 1]
```diff
@@ -131,6 +131,7 @@ local frecency = function(opts)
       if state.active_filter_tag == "LSP" or state.active_filter_tag == "CWD" then
         filter_path = utils.path_tail(state.active_filter) .. os_path_sep
       else
+        filter_path = Path:new(state.active_filter):make_relative(os_home) .. os_path_sep
         filter_path = path.make_relative(state.active_filter, os_home) .. os_path_sep
```
Comment (clason) [inline lua/telescope/_extensions/frecency.lua:135]: should be removed, then?

File: lua/telescope/_extensions/frecency.lua [hunk 2]
```diff
@@ -37,12 +37,12 @@ local function format_filepath(filename, opts)
   local original_filename = filename
 
   if state.active_filter then
-    filename = path.make_relative(filename, state.active_filter)
+    filename = Path:new(filename):make_relative(state.active_filter)
   else
-    filename = path.make_relative(filename, state.cwd)
+    filename = Path:new(filename):make_relative(cwd)
```
Comment (clason) [inline lua/telescope/_extensions/frecency.lua:42]: should be `state.cwd`?

**Comment by Conni2461** [comment]
Good call thanks :)

**Comment by clason** [comment]
Yep, looks good and works, thanks!

**Comment by Conni2461** [comment]
Thank you for always being here :) and catching these things

**Comment by clason** [comment]
No problem; least I can do :)

**Pull request merged**